servers/shares-api/src/apollo/context.ts

```diff
@@ -25,32 +26,46 @@ export async function getContext({
 
 export interface IContext {
   PocketShareModel: PocketShareModel;
+  User: UserContext;
 }
 
 export class ContextManager implements IContext {
   PocketShareModel: PocketShareModel;
+  User: UserContext;
   constructor(options: { request: Request }) {
-    const userId = options.request.headers.userid;
+    const rawUserId = options.request.headers.userid;
+    const rawGuid = options.request.headers.guid;
     const isNative =
       options.request.headers.applicationisnative === 'true' ? true : false;
     const client = dynamoClient();
+    // We have an authenticated user?
```

Comment:
Weeee! so many conditionals.... prob should be a package... at some point :p

Reply:
Yeah would be good to just have a JWT user model that's shared